tests/functional/afr/test_self_heal_with_full_algorithm.py

```diff
@@ -0,0 +1,207 @@
+           'distributed-replicated'], ['glusterfs']])
+class TestSelfHealWithFullAlgorithm(GlusterBaseClass):
+    def setUp(self):
+
+        # Calling GlusterBaseClass setUp
+        self.get_super_method(self, 'setUp')()
+
+        if not self.setup_volume_and_mount_volume(self.mounts):
+            raise ExecutionError("Unable to setup and mount volume")
+        g.log.info("Volume created and mounted successfully")
+
+    def tearDown(self):
+
+        # Unmount and cleanup the volume
+        if not self.unmount_volume_and_cleanup_volume(self.mounts):
+            raise ExecutionError("Unable to unmount and cleanup volume")
+        g.log.info("Volume unmounted and deleted successfully")
+
+        # Calling GlusterBaseClass Teardown
+        self.get_super_method(self, 'tearDown')()
+
+    def test_self_heal_with_full_algorithm(self):
+        """
+        Test Steps:
+        1. Create a replicated/distributed-replicate volume and mount it
+        2. Set data/metadata/entry-self-heal to off and
+           data-self-heal-algorithm to full
+        3. Create a directory from the mount point
+        4. Bring down all bricks from a selected set
+        5. Create few files inside the directory with some data
+        6. Bring back the bricks online and wait for heal to complete
+        7. Check arequal of the subvol and all the bricks in the subvol should
+           have same checksum
+        8. Bring down all bricks from another set
+        9. Modify the data of existing files under the directory
+        10. Bring back the bricks online and wait for heal to complete
+        11. Check arequal of the subvol and all the brick in the same subvol
+            should have same checksum
+        """
+
+        # Setting options
+        for key, value in (("data-self-heal", "off"),
+                           ("metadata-self-heal", "off"),
+                           ("entry-self-heal", "off"),
+                           ("data-self-heal-algorithm", "full")):
+            ret = set_volume_options(self.mnode, self.volname, {key: value})
+            self.assertTrue(ret, 'Failed to set %s to %s.' % (key, value))
+            g.log.info("%s set to %s successfully", key, value)
+
+        # Create a directory under the mount point
+        mountpoint = self.mounts[0].mountpoint
+        client = self.mounts[0].client_system
+
+        cmd = ("mkdir %s/test_full_self_heal" % (mountpoint))
+        ret, _, _ = g.run(client, cmd)
+        self.assertEqual(ret, 0, "Failed to create directory on mountpoint")
+        g.log.info("Successfully created directory on mountpoint")
+
+        # List a brick in each subvol and bring them offline
+        brick_to_bring_offline = []
+        subvols = get_subvols(self.mnode, self.volname)['volume_subvols']
+        for subvol in subvols:
+            self.assertTrue(subvol, "List is empty")
+            brick_to_bring_offline.extend(sample(subvol, 1))
```

Comment:
I'm guessing even if we bring arbiter brick down, functionality doesn't change in current context?

Reply:
Yes, it won't change the functionality. But bringing arbiter down won't add any value to the test, since it doesn't store any data in it. So even if we do any data changes after bringing the arbiter brick down, the checksums will still match on the data brick even before the heal.

Reply:
requires fix

Reply:
Since arbiter won't have any data and we are checking specifically for data heal, not bringing down the arbiter brick(s). Changed the logic to bring down the specific bricks in the replica subvols down instead of choosing one randomly.